lib/ansible/galaxy/collection.py

```diff
@@ -1384,7 +1384,19 @@ def _download_file(url, b_path, expected_hash, validate_certs, headers=None):
 
 def _extract_tar_dir(tar, dirname, b_dest):
     """ Extracts a directory from a collection tar. """
-    tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+    try:
+        tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+    except KeyError:
+        # Some archives contain directories that end with '/'
+        if not dirname.endswith(os.path.sep):
+            dirname += os.path.sep
+
+        try:
+            tar_member = tar.getmember(to_native(dirname, errors='surrogate_or_strict'))
+        except KeyError:
+            # If we still can't find the member to extract, raise a nice error.
+            raise AnsibleError("Unable to extract '%s' from collection" % to_native(dirname))
```

Comment:
This should probably mention both slash and no-slash paths...

Reply:
Since it seems this archive with the trailing `/` is an anomaly, I don't think we nee to mention both versions of the path.